### PR TITLE
Hide section picker if when viewing column settings

### LINKED ISSUE #10492: When viewing column settings in viz options we should hide the other sidebar "chrome"
Right now it looks confusing after you click into a columns settings. The column back link should be at the top, and we should hide the tabs and the back button to go back to the viz type selection panel.

column-options

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #10492

New behavior:
hide-section-picker

### CODE REVIEW

**Comment by mazameli** [comment]
Just looking at the GIF, this doesn't do quite what I'd suggested. I think that once you click on the column settings gear, not only should the bubble tabs go away, but the top breadcrumb link should display the column name _instead of_ the `[Chart type] options` text. I think it's weird to have two back breadcrumb links at the top that go to different destinations.

**Comment by paulrosenzweig** [comment]
New new behavior:

hide-section-picker

@mazameli is that right or should the only breadcrumb be styled as a top-level one? I was going off this:

image


Another small change I made was removing logic to hide the column breadcrumb if there's only one column. I'm not sure why we had that to begin with, so maybe I'm missing something.

**Comment by mazameli** [comment]
> Should the only breadcrumb be styled as a top-level one?

Seeing it in practice, I do think it would feel nicer if the column breadcrumb was styled as a top-level one like you said (i.e. the black text, and larger). Ideally it would also appear in exactly the same position as the visualization type breadcrumb does so that things don't shift around after you click the gear.

> Another small change I made was removing logic to hide the column breadcrumb if there's only one column. I'm not sure why we had that to begin with, so maybe I'm missing something.

Weird, yeah, I'm not sure why that would have been there.

**Comment by paulrosenzweig** [comment]
@mazameli, how does this look?

hide-section-picker

**Comment by mazameli** [comment]
Great!

image

File: frontend/src/metabase/query_builder/components/SidebarContent.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default function SidebarContent({
   className,
   icon,
   title,
+  titleOverride,
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/SidebarContent.jsx]: I don't understand why `SidebarContent` needs a `titleOverride` prop. Can't you just do something like

```
<SidebarContent
  title={sidebarTitleOverride  ? sidebarTitleOverride.title : "normal title"}
  onBack={sidebarTitleOverride ? sidebarTitleOverride.onBack : whatever}'
>
```

or even 

```
<SidebarContent
  title="normal title"
  onBack={whatever}
  {...(sidebarTitleOverride || {})}
>
```

Maybe call it `sidebarPropsOverride` too.

**Review by tlrobinson — CHANGES_REQUESTED** [review]

File: frontend/src/metabase/query_builder/components/SidebarContent.jsx (hunk 1 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/query_builder/components/SidebarContent.jsx]: much better idea! I'll change it to `sidebarPropsOverride`

Comment (paulrosenzweig) [reply frontend/src/metabase/query_builder/components/SidebarContent.jsx]: updated

File: frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx [hunk 2]
```diff
@@ -5,38 +5,49 @@ import ChartSettings from "metabase/visualizations/components/ChartSettings";
...
-        ]}
-        onChange={onReplaceAllVisualizationSettings}
-        onClose={onClose}
-        noPreview
-        initial={initialChartSetting}
-      />
-    </SidebarContent>
-  );
+export default class ChartSettingsSidebar extends React.Component {
+  state = { sidebarPropsOverride: undefined };
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx]: FYI typically I'd use `null` here, but I can't really articulate why at the moment :)

**Review by tlrobinson — APPROVED** [review]

File: frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx (hunk 2 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx]: yeah that's fair. I guess it's kinda weird to _define_ something as `undefined`.

File: frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx [hunk 3]
```diff
@@ -5,38 +5,49 @@ import ChartSettings from "metabase/visualizations/components/ChartSettings";
...
+    } = this.props;
+    const { sidebarPropsOverride } = this.state;
+    return (
+      result && (
+        <SidebarContent
+          className="full-height"
+          title={t`${visualizations.get(question.display()).uiName} options`}
+          onDone={onClose}
+          onBack={onOpenChartType}
+          {...sidebarPropsOverride}
```
Comment (paulrosenzweig) [inline frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx]: Apparently  `{...null}` is fine! And, so is `{...undefined}`, `{...false}`, etc.

I guess that's consistent with `false.foo` not throwing an error. It just has no properties to spread, but there's nothing wrong with asking. 🤷‍♂

**Pull request merged**